src/libraries/System.Text.Json/src/System/Text/Json/Serialization/ReflectionEmitMemberAccessor.cs

```diff
@@ -294,8 +291,12 @@ private static Delegate CreatePropertySetter(PropertyInfo propertyInfo, Type cla
 
             generator.Emit(OpCodes.Ret);
 
-            return dynamicMethod.CreateDelegate(typeof(Action<,>).MakeGenericType(objectType, propertyType));
+            return dynamicMethod;
         }
+
+        [return: NotNullIfNotNull("method")]
+        private static T? CreateDelegate<T>(DynamicMethod? method) where T : Delegate =>
```

Comment:
Would we still need the code sharing after generic `CreateDelegate<T>` methods are added to `MethodInfo`?

Reply:
Answering on the first version of the comment.

This is a helper method only whch I made to not repeat `(SomeDelegate)CreateMethod(...).CreateDelegate(typeof(SomeDelegate))` many time in the file. But what's about sharing? There is [a runtime feature called shared generics](docs/design/coreclr/botr/shared-generics.md), but it works only if all type parameters are reference types. If not, the JIT will produce a method implementation for each unique set of parameters.

Therefore, you easily could get a lot of jitted `CreateParameterizedConstructor<T>` which will be used once. This is pure code pollution (: But having a common version which doesn't use generics helps to reduce memory pressure, so there will be a lot of small `CreateParameterizedConstructor<T>` calling the common version with all required parameters obtained from their type params.